src/pybamm/expression_tree/operations/serialise.py

```diff
@@ -399,3 +604,315 @@ def _convert_options(self, d):
             return tuple(self._convert_options(item) for item in d)
         else:
             return d
+
+    @staticmethod
+    def convert_symbol_to_json(symbol):
+        """
+        Recursively converts a PyBaMM symbolic expression into a JSON-serializable format.
+
+        Supports most PyBaMM symbol types, including scalars, variables, parameters,
+        operators, broadcasts, and interpolants.
+
+        Parameters
+        ----------
+        symbol : pybamm.Symbol or compatible type
+            The expression or object to convert.
+
+        Returns
+        -------
+        dict
+            A JSON-compatible representation of the input.
+
+        Examples
+        --------
+        >>> import pybamm
+        >>> from pybamm.expression_tree.operations.serialise import Serialise
+        >>> s = pybamm.Scalar(5)
+        >>> Serialise.convert_symbol_to_json(s)
+        {'type': 'Scalar', 'value': 5}
```

Comment:
Is this 5.0 (float value) expected?
```suggestion
        {'type': 'Scalar', 'value': 5.0}
```